examples/reporting/campaign_report_to_csv.rb

```diff
@@ -0,0 +1,136 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This code example illustrates how to get metrics about a campaign and 
+# serialize the result as a CSV file.
+
+require 'optparse'
+require 'time'
+require 'csv'
+require 'google/ads/google_ads'
+
+PAGE_SIZE = 1000
+
+def result_row_as_csv_hash(result_row)
+  {
+    id: result_row.campaign.id.value,
+    name: result_row.campaign.name.value,
+    date: Time.parse(result_row.segments.date.value),
+    impressions: result_row.metrics.impressions.value,
+    clicks: result_row.metrics.clicks.value,
+    cost_micros: result_row.metrics.cost_micros.value,
+  }
+end
+
+def write_campaign_report_csv(customer_id, target_filepath)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  ga_service = client.service(:GoogleAds)
+
+  query = <<~QUERY
+      SELECT campaign.id,
+             campaign.name,
+             segments.date,
+             metrics.impressions,
+             metrics.clicks,
+             metrics.cost_micros
+      FROM campaign
+      WHERE segments.date DURING LAST_7_DAYS
+        AND campaign.status = 'ENABLED'
+      ORDER BY segments.date DESC
+  QUERY
+
+  response = ga_service.search(customer_id, query, page_size: PAGE_SIZE)
+  # convert the Google Ads response rows in to CSV ready hash objects
+  csv_rows = response.map { |row| result_row_as_csv_hash(row) }
+
+  CSV.open(target_filepath, "wb") do |csv|
+    # use the keys of the first csv_row as a header row
+    csv << csv_rows.first.keys
+
+    # write all the values as rows of the CSV
+    csv_rows.each do |row|
+      csv << row.values
+    end
+  end
+
+  puts "successfully created #{target_filepath} with #{csv_rows.count} entries"
+end
+
+if __FILE__ == $PROGRAM_NAME
+  options = {}
+
+  # The following parameter(s) should be provided to run the example. You can
+  # either specify these by changing the INSERT_XXX_ID_HERE values below, or on
+  # the command line.
+  #
+  # Parameters passed on the command line will override any parameters set in
+  # code.
+  #
+  # Running the example with -h will print the command line usage.
+  options[:customer_id] = 'INSERT_ADWORDS_CUSTOMER_ID_HERE'
+
+  OptionParser.new do |opts|
+    opts.banner = sprintf('Usage: ruby %s [options]', File.basename(__FILE__))
+
+    opts.separator ''
+    opts.separator 'Options:'
+
+    opts.on('-C', '--customer-id CUSTOMER-ID', String, 'Customer ID') do |v|
+      options[:customer_id] = v
+    end
+
+    opts.separator ''
+    opts.separator 'Help:'
+
+    opts.on_tail('-h', '--help', 'Show this message') do
+      puts opts
+      exit
+    end
+  end.parse!
+
+  begin
+    # write CSV output to campaign_report_to_csv.csv
```

Comment:
I think it would be nice to allow the user to specify a filepath on the command line as well.